NETDefault/UnitTestingWithNUnit&Moq/UserServiceTests.cs

```diff
@@ -9,65 +11,54 @@ internal class UserServiceTests : UserServiceTestsBase
         public void GetDashboard_WhenCustomerNameNotFound_ItShouldThrow()
         {
             // Arrange
-            SetupGetUserIdByUserName(_unknownUsername);
+            SetupGetUserIdByUserName();
 
             // Assert
-            Assert.Throws<NotFoundException>(() => _subject.GetDashboardByUserName(_unknownUsername));
-
-            _gateway.Verify(x => x.GetUserIdByUserName(_unknownUsername), Times.Once());
-            _gateway.VerifyNoOtherCalls();
+            Assert.Throws<NotFoundException>(() => _subject.GetDashboardByUserName(UnknownUsername));
         }
 
         [Test]
         public void GetDashboard_WhenCustomerNameFound_ItShouldBeReturned()
         {
             // Arrange
-            SetupGetUserIdByUserName(_knownUsername);
+            SetupGetUserIdByUserName();
 
-            var expected = _userId.ToString();
+            var expected = UserId.ToString();
```

Comment:
I would suggest on not depending on `const`s as expected values in unit test. I rather have a hard coded string in the assert, than having this. This assert is also pretty weak (just find a number somewhere), so perhaps refactor that also a bit to make this example a little better.

Reply:
It are now local constants where we use the value multiple times

Reply:
The expected value is still the same one as used in setting up the mock. That's not great in my opinion. Lets way we set this UserId to 0 and the test still works. Then we break the service by always returning 0, and the test still works. I would like to see the `expected` be something independent of the mocked behavior.

Reply:
mock and expected are now different values.